driver-core/src/test/functional/com/mongodb/client/model/expressions/ExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,646 @@
+        );
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/meta/
+    }
+
+
+    @Test
+    public void comparisonExpressionsTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#comparison-expression-operators
+        // (Complete as of 6.0)
+        // Comparison expressions are on the generic Expression class.
+
+        BooleanExpression tru = Expressions.ofBoolean(true);
+        StringExpression str = Expressions.ofString("string");
+        Expression nul = Expressions.ofNull();
+
+        // equality:
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/eq/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/ne/
+
+        // comparison:
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/gt/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/gte/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/lt/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/lte/
+
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/cmp/ (89)
+        // TODO implement cmp as a three-branch cond?
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/ifNull/
+        // This is similar to orElse, but treats nulls as empty Optionals
+        // TODO the last position is auto-filled with null
+        // TODO this diverges from the MQL
+        // TODO just as with 1.add(2).add(3), multi-param ifNull should be chained: a.ifNull(b).ifNull(c)
+
+        assertExpression(
+                true,
+                nul.ifNull(tru).ifNull(str),
+                // MQL
+                "{'$ifNull': [{'$ifNull': [null, true, null]}, 'string', null]}");
+        assertExpression(
+                "string",
+                nul.ifNull(nul).ifNull(str),
+                // MQL
+                "{'$ifNull': [{'$ifNull': [null, null, null]}, 'string', null]}");
+    }
+
+
+
+
+
+    @Test
+    public void arithmeticExpressionsTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#arithmetic-expression-operators
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/add/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/sum/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/multiply/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/subtract/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/divide/
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/max/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/min/ (63)
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/round/
+
+        // TODO helper for bound constrain via max/min?
```

Comment:
There are a number of places where we can introduce helper methods. IIRC we have done this for full text search, by adding something like `gtlt`. But if we add these to the API, they should really be added to the server - but that is much more difficult to do.

Reply:
This was generally not done. If it is important enough to do, we or customers will realize this in the future.